### DISCUSSION TITLE
Aggregator: How can get parameter from first service and then use a data from response of the service as a parameter to next service?

### DISCUSSION THREAD

**Discussion by RameshkumarJay** [ThreeMammals/Ocelot#1931] [Q&A] [answered]
My requirement is I receive a parameter from client and pass it to my first downstream service and then use a data from response of the downstream service as a parameter to the next downstream service.

Is it possible or not by using Ocelot? Please, provide one sample example by using .NET core with Ocelot.
I am new in microserices.
Thanks in Advance.

**Comment by raman-m** [comment, marked as answer]
Impossible now!
You have described a kind of sequential process which can be started by first request.
This workflow should be defined somewhere.
So, Ocelot route syntax doesn't allow to link steps to define a workflow.
But in theory, everything is possible.
For now, you can define a route of first step.
Then, we have to define other steps, right?
I guess, the most appropriate place is custom [Delegating Handler](https://ocelot.readthedocs.io/en/latest/features/delegatinghandlers.html).
In the handler define the rest of your workflow steps, and link them.
As a result, return last step response from the handler back to upstream.

### FYI
Aggregator feature is not suitable place to define workflow steps, because the function of aggregator is combining all responses as one JSON response.
Technically, in theory, workflow steps can be defined even in custom aggregator class. But design will be a bit ugly.
